File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
from collections import defaultdict

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        def my_intuition(nums, k):
            count_map = defaultdict(int)

            for i in nums:
                count_map[i] += 1

            counts = count_map.items()

            sorted_counts  = sorted(counts, key=lambda x:x[1], reverse=True)

            res = []
            for j in sorted_counts[:k]:
                res.append(j[0])
            return res

        return my_intuition(nums, k)



        

File: Data Structures & Algorithms/binary-search/submission-0.cpp
class Solution {
public:
    int binary_search_(int l, int r, vector<int>& nums, int target){
        if (l > r) return -1;
        int m = l + (r - l) / 2;

        if (nums[m] == target) return m;

        if (target < nums[m]) {
            return binary_search_(l, m-1, nums, target);
        } else {
            return binary_search_(m+1, r, nums, target);
        }

    }

    int search(vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size()-1;
        return binary_search_(l, r, nums, target);
    }
};
